src/lens.jl

```diff
@@ -16,78 +16,87 @@ julia> using Setfield
 
 julia> struct T;a;b; end
 
-julia> t = T("AA", "BB")
+julia> obj = T("AA", "BB")
 T("AA", "BB")
 
-julia> l = @lens _.a
+julia> lens = @lens _.a
 (@lens _.a)
 
-julia> get(l, t)
+julia> get(obj, lens)
 "AA"
 
-julia> set(l, t, 2)
+julia> set(obj, lens, 2)
 T(2, "BB")
 
-julia> t
+julia> obj
 T("AA", "BB")
 
-julia> modify(lowercase, l, t)
+julia> modify(lowercase, obj, lens)
 T("aa", "BB")
 ```
 
 # Interface
 Concrete subtypes of `Lens` have to implement
-* `set(lens, obj, val)`
-* `get(lens, obj)`
+* `set(obj, lens, val)`
+* `get(obj, lens)`
 
 These must be pure functions, that satisfy the three lens laws:
-* `get(lens, set(lens, obj, val)) == val` (You get what you set.)
-* `set(lens, obj, get(lens, obj)) == obj` (Setting what was already there changes nothing.)
-* `set(lens, set(lens, obj, val1), val2) == set(lens, obj, val2)` (The last set wins.)
+
+```jldoctest; output = false, setup = :(using Setfield; obj = (a="A", b="B"); lens = @lens _.a; val = 2; val1 = 10; val2 = 20)
```

Comment:
It is a bit ugly but doctest is the easiest way to make sure I don't make any typos in the sample code...

In REPL, `?Lens` looks fine:

![image](https://user-images.githubusercontent.com/29282/44897231-d6ee5d00-acaf-11e8-8225-bc9410520257.png)

(except `# output` part)

HTML rendering is a bit better:

![image](https://user-images.githubusercontent.com/29282/44897260-eff70e00-acaf-11e8-94f4-a8f900782488.png)